en: fix typos & add release notes (#2839)
pingcap/docs-tidb-operator: en/releases/release-2.0.0-beta.0.md

For a detailed comparison between v2 and v1, see [Comparison Between TiDB Operator v2 and v1](../tidb-operator-v2-vs-v1.md).

> **Warning:**
>
> This is a beta release. **Test thoroughly before deploying in production environments.**

## Major changes and improvements

### Core architecture refactoring

TiDB Operator v2 includes a comprehensive redesign of the v1 architecture, with the following key changes:

- **CRD splitting**: split the v1 `TidbCluster` CRD into multiple independent CRDs for more granular component management, improving maintainability and flexibility.
- **Direct Pod management**: remove the dependency on StatefulSet. Pods are now managed directly, providing higher flexibility and more precise control over Pod lifecycle and scheduling behavior.
- **Controller architecture upgrade**: implement controller logic based on the [controller-runtime](https://github.com/kubernetes-sigs/controller-runtime) framework. This simplifies controller development, improves development efficiency, and enhances system stability and reliability.

### New features and enhancements

- **Support the Overlay field**:

>     - Enable you to flexibly specify all Kubernetes-supported fields for Pods without modifying TiDB Operator source code.

Reviewer: ```suggestion
    - Enable you to flexibly specify all Kubernetes-supported fields for Pods without modifying the TiDB Operator source code.
```

    - Provide security validation mechanisms to prevent accidental overwrites of critical system labels.

- **Support topology-aware scheduling**:
    - Support the `EvenlySpread` strategy to evenly distribute Pods across topology domains.
    - Support topology weight configuration for flexible control of instance distribution ratios across topology domains.
    - Enhance cluster high availability and fault tolerance.

- **Enhance the field validation**:
    - Integrate Kubernetes [Validation Rules](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#validation-rules) and [Validating Admission Policy](https://kubernetes.io/docs/reference/access-authn-authz/validating-admission-policy/).
    - Support field format and value range validation.
    - Provide clear and user-friendly error messages to facilitate troubleshooting.

- **Support [CRD subresources](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#subresources)**:
    - Support the `status` subresource for unified status management.

>     - Support the `scale` subresource for integration with [HorizontalPodAutoscaler (HPA)](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/), enabling auto-scaling.

Reviewer: ```suggestion
    - Support the `scale` subresource to integrate with [HorizontalPodAutoscaler (HPA)](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/), enabling auto-scaling.
```

    - Improve compatibility with the Kubernetes ecosystem.

> - **Optimize the configuration management**:
>     - Optimize the configuration hash algorithm to avoid unnecessary rolling updates caused by ineffective changes.

Reviewer: ```suggestion
- **Optimize the configuration management**:
    - Optimize the configuration hash algorithm to avoid unnecessary rolling updates caused by invalid changes.
```
